lib/src/Base/Func/DistanceToDomainEvaluation.cxx

```diff
@@ -0,0 +1,115 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief The class that implements the composition between numerical
+ *        math functions implementations
+ *
+ *  Copyright 2005-2021 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "openturns/DistanceToDomainEvaluation.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/Interval.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+typedef Collection<UnsignedInteger>     UnsignedIntegerCollection;
+
+CLASSNAMEINIT(DistanceToDomainEvaluation)
+
+static const Factory<DistanceToDomainEvaluation> Factory_DistanceToDomainEvaluation;
+
+/* Default constructor */
+DistanceToDomainEvaluation::DistanceToDomainEvaluation()
+  : EvaluationImplementation()
+  , domain_(Interval())
+{
+  // Nothing to do
+} // DistanceToDomainEvaluation
+
+/* Default constructor */
+DistanceToDomainEvaluation::DistanceToDomainEvaluation(const Domain & domain)
+  : EvaluationImplementation()
+  , domain_(domain)
+{
+  EvaluationImplementation::setInputDescription(Description::BuildDefault(domain_.getDimension(), "x"));
+} // DistanceToDomainEvaluation
+
+/* Virtual constructor */
+DistanceToDomainEvaluation * DistanceToDomainEvaluation::clone() const
+{
+  return new DistanceToDomainEvaluation(*this);
+}
+
+
+/* Comparison operator */
+Bool DistanceToDomainEvaluation::operator ==(const DistanceToDomainEvaluation & ) const
+{
+  return true;
```

Comment:
@regislebrun mentioned yesterday that this is incorrect. Since it is copied from `IndicationEvaluation::operator ==`, the original should also be changed.

Could we perhaps do it this way?

```suggestion
Bool DistanceToDomainEvaluation::operator ==(const DistanceToDomainEvaluation & other) const
{
  return domain_ == other.getDomain();
```
Of course, to do this, I also need to implement the accessor `getDomain()`.

Reply:
not necessarily, you can access the attribute from there

Reply:
OK. Done.